### PR TITLE
doc: translate chapter_tree/binary_tree_traversal.md

### PR DESCRIPTION
**Pull request by QiLOL**
If this pull request (PR) pertains to **Chinese-to-English translation**, please confirm that you have read the contribution guidelines and complete the checklist below:

- [x] This PR represents the translation of a single, complete document, or contains only bug fixes.
- [x] The translation accurately conveys the original meaning and intent of the Chinese version. If deviations exist, I have provided explanatory comments to clarify the reasons.

If this pull request (PR) is associated with **coding or code transpilation**, please attach the relevant console outputs to the PR and complete the following checklist:

- [ ] I have thoroughly reviewed the code, focusing on its formatting, comments, indentation, and file headers.
- [ ] I have confirmed that the code execution outputs are consistent with those produced by the reference code (Python or Java).
- [ ] The code is designed to be compatible on standard operating systems, including Windows, macOS, and Ubuntu.

### CODE REVIEW

File: en/docs/chapter_tree/binary_tree_traversal.md [hunk 1]
```diff
@@ -22,14 +22,14 @@ Breadth-first traversal is usually implemented with the help of a "queue". The q
 
 ### Complexity analysis
 
-- **Time complexity is $O(n)$**: All nodes are visited once, using $O(n)$ time, where $n$ is the number of nodes.
-- **Space complexity is $O(n)$**: In the worst case, i.e., a full binary tree, before traversing to the lowest level, the queue can contain at most $(n + 1) / 2$ nodes at the same time, occupying $O(n)$ space.
+- **Time complexity is $O(n)$**: All nodes are visited once, taking $O(n)$ time, where $n$ is the number of nodes.
+- **Space complexity is $O(n)$**: In the worst case, i.e., a full binary tree, before traversing to the bottom level, the queue can contain at most $(n + 1) / 2$ nodes simultaneously, occupying $O(n)$ space.
```
Comment (pengchzn) [inline en/docs/chapter_tree/binary_tree_traversal.md:26]: Maybe **lowest level** will be better

You can check its usage [here](https://echobind.com/post/binary-trees-what-are-they)

File: en/docs/chapter_tree/binary_tree_traversal.md [hunk 2]
```diff
@@ -1,14 +1,14 @@
 # Binary tree traversal
 
-From the perspective of physical structure, a tree is a data structure based on linked lists, hence its traversal method involves accessing nodes one by one through pointers. However, a tree is a non-linear data structure, which makes traversing a tree more complex than traversing a linked list, requiring the assistance of search algorithms to achieve.
+From a physical structure perspective, trees are data structures based on linked lists. Hence, their traversal method involves accessing nodes one by one through pointers. However, trees are non-linear data structures, which makes traversing a tree more complex than traversing a linked list, requiring the assistance of search algorithms.
```
Comment (pengchzn) [inline en/docs/chapter_tree/binary_tree_traversal.md:3]: 'From a physical structure perspective, trees are data structures based on linked lists.' --> 'From a physical structure perspective, a tree is a linked-list-based data structure.'

"Tree" refers as a data structure, using the plural form feels incorrect

Comment (pengchzn) [reply en/docs/chapter_tree/binary_tree_traversal.md:3]: 'their' --> 'its'

Comment (pengchzn) [reply en/docs/chapter_tree/binary_tree_traversal.md:3]: This paragraph is discussing one specific data structure—a tree. It is not referring to multiple trees or instances of trees, so maybe the singular form should be sufficient.

File: en/docs/chapter_tree/binary_tree_traversal.md (hunk 1 above)
Comment (QiLOL) [reply en/docs/chapter_tree/binary_tree_traversal.md:26]: Thank you, I would keep this one based on [ref](https://www.andrew.cmu.edu/course/15-121/lectures/Trees/trees.html)

File: en/docs/chapter_tree/binary_tree_traversal.md (hunk 2 above)
Comment (QiLOL) [reply en/docs/chapter_tree/binary_tree_traversal.md]: Thanks, updated.

**Review by krahets — APPROVED** [review]
Thanks, @QiLOL @pengchzn !

**Pull request merged**